src/app/3d/qgspointcloud3dsymbolwidget.cpp

```diff
@@ -0,0 +1,81 @@
+/***************************************************************************
+  qgspointcloud3dsymbolwidget.cpp
+  ------------------------------
+  Date                 : November 2020
+  Copyright            : (C) 2020 by Nedjima Belgacem
+  Email                : belgacem dot nedjima at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgspointcloud3dsymbolwidget.h"
+
+#include "qgspointcloudlayer.h"
+#include "qgspointcloud3dsymbol.h"
+#include "qgspointcloudlayer3drenderer.h"
+
+QgsPointCloud3DSymbolWidget::QgsPointCloud3DSymbolWidget( QgsPointCloudLayer *layer, QWidget *parent )
+  : QWidget( parent )
+  , mLayer( layer )
+{
+  this->setupUi( this );
+
+  if ( layer )
+  {
+    QgsPointCloudLayer3DRenderer *renderer = dynamic_cast<QgsPointCloudLayer3DRenderer *>( layer->renderer3D() );
+    QgsPointCloud3DSymbol *symbol;
+    if ( renderer != nullptr )
+    {
+      symbol = const_cast<QgsPointCloud3DSymbol *>( renderer->symbol() );
+      setSymbol( symbol );
+    }
+    else
+    {
+      symbol = new QgsPointCloud3DSymbol;
+      setSymbol( symbol );
+      delete symbol;
+    }
+  }
+
+  connect( mPointSizeSpinBox, qgis::overload<double>::of( &QDoubleSpinBox::valueChanged ), this, &QgsPointCloud3DSymbolWidget::changed );
+}
+
+void QgsPointCloud3DSymbolWidget::setLayer( QgsPointCloudLayer *layer )
+{
+  mLayer = layer;
+  if ( layer )
+  {
+    QgsPointCloudLayer3DRenderer *renderer = dynamic_cast<QgsPointCloudLayer3DRenderer *>( layer->renderer3D() );
+    QgsPointCloud3DSymbol *symbol;
+    if ( renderer != nullptr )
+    {
+      symbol = const_cast<QgsPointCloud3DSymbol *>( renderer->symbol() );
+      setSymbol( symbol );
+    }
+    else
+    {
+      symbol = new QgsPointCloud3DSymbol;
+      setSymbol( symbol );
+      delete symbol;
+    }
+  }
+}
+
+void QgsPointCloud3DSymbolWidget::setSymbol( QgsPointCloud3DSymbol *symbol )
+{
+  mPointSizeSpinBox->setValue( symbol->pointSize() );
+}
+
+QgsPointCloud3DSymbol *QgsPointCloud3DSymbolWidget::symbol() const
+{
+  // TODO: fix memory leak
```

Comment:
This isn't a leak anymore (if you make the suggested change of having the renderer take ownership of the symbol)